code/modules/surgery/organs/autosurgeon.dm

```diff
@@ -3,92 +3,108 @@
 	icon = 'icons/obj/device.dmi'
-	icon_state = "autoimplanter"
+	icon_state = "autosurgeon"
 	inhand_icon_state = "nothing"
 	w_class = WEIGHT_CLASS_SMALL
 
+	//how many times you can use the autosurgeon before it becomes useless
 	var/uses = INFINITE
-
-/obj/item/autosurgeon/attack_self_tk(mob/user)
-	return //stops TK fuckery
-
-/obj/item/autosurgeon/organ
-	name = "implant autosurgeon"
-	desc = "A device that automatically inserts an implant or organ into the user without the hassle of extensive surgery. \
-		It has a slot to insert implants or organs and a screwdriver slot for removing accidentally added items."
-
-	var/organ_type = /obj/item/organ
+	// What organ will the subtype start with
 	var/starting_organ
+	// The organ currently loaded in the autosurgeon
 	var/obj/item/organ/storedorgan
+	// list of organs and their children we allow into the autosurgeon- an empty list means no whitelist
+	var/list/organ_whitelist = list()
+	// percentage modifier for how fast the surgery happens on other people
 	var/surgery_speed = 1
+	// overlay that shows when the autosurgeon has a stored organ
+	var/loaded_overlay = "autosurgeon_loaded_overlay"
 
-/obj/item/autosurgeon/organ/syndicate
-	name = "suspicious implant autosurgeon"
-	icon_state = "syndicate_autoimplanter"
-	surgery_speed = 0.75
+/obj/item/autosurgeon/attack_self_tk(mob/user)
+	return //stops TK fuckery
 
-/obj/item/autosurgeon/organ/Initialize(mapload)
+/obj/item/autosurgeon/Initialize(mapload)
 	. = ..()
 	if(starting_organ)
-		insert_organ(new starting_organ(src))
+		load_organ(new starting_organ(src))
+
+/obj/item/autosurgeon/proc/load_organ(obj/item/organ/loaded_organ, mob/living/user)
+	if(user)
+		if(storedorgan)
+			to_chat(user, span_alert("[src] already has an implant stored."))
+			return
 
-/obj/item/autosurgeon/organ/proc/insert_organ(obj/item/item)
-	storedorgan = item
-	item.forceMove(src)
-	name = "[initial(name)] ([storedorgan.name])"
+		if(uses == 0)
+			to_chat(user, span_alert("[src] is used up and cannot be loaded with more implants."))
+			return
 
-/obj/item/autosurgeon/organ/proc/use_autosurgeon()
-	storedorgan = null
-	name = initial(name)
-	if(uses != INFINITE)
-		uses--
-	if(!uses)
-		desc = "[initial(desc)] Looks like it's been used up."
+		if(organ_whitelist.len)
+			var/obj/item/organ/organ_whitelisted
+			for(var/whitelisted_organ in organ_whitelist)
+				if(istype(loaded_organ, whitelisted_organ))
+					organ_whitelisted = TRUE
+					break
+			if(!organ_whitelisted)
+				to_chat(user, span_alert("[src] is not compatible with [loaded_organ]."))
+				return
+
+		if(!user.transferItemToLoc(loaded_organ, src))
+			to_chat(user, span_alert("[loaded_organ] is stuck to your hand!"))
+			return
+
+	storedorgan = loaded_organ
+	loaded_organ.forceMove(src)
+	name = "[initial(name)] ([storedorgan.name])" //to tell you the organ type, like "suspicious autosurgeon (Reviver implant)"
+	add_overlay(loaded_overlay)
```

Comment:
rather then managing the overlay like this, you should make it part of update_appearance
also should make it emissive, that'll look quite nice